src/Microsoft.OData.Core/JsonLight/ODataJsonLightBatchReader.cs

```diff
@@ -167,6 +167,14 @@ protected override ODataBatchOperationRequestMessage CreateOperationRequestMessa
             string url = (string)this.messagePropertiesCache.GetPropertyValue(
                 ODataJsonLightBatchPayloadItemPropertiesCache.PropertyNameUrl);
             ValidateRequiredProperty(url, ODataJsonLightBatchPayloadItemPropertiesCache.PropertyNameUrl);
+
+            // escape any colons in the query string portion of the url
+            int queryOptionSeparator = url.IndexOf("?", StringComparison.OrdinalIgnoreCase);

```

Comment:
>url [](start = 39, length = 3)

what happen if the key string has "?" ?

Reply:
If the key string has a ? it must be escaped.

According to [https://tools.ietf.org/html/rfc2396](https://tools.ietf.org/html/rfc2396), section 3.3, "Within a path segment, the characters "/", ";", "=", and "?" are reserved."


---
In reply to: [279941429](https://github.com/OData/odata.net/pull/1457#discussion_r279941429) [](ancestors = 279941429)